core/src/test/scala/unit/kafka/log/LocalLogTest.scala

```diff
@@ -362,8 +363,8 @@ class LocalLogTest {
     }
     assertEquals(5, log.segments.numberOfSegments)
     assertNotEquals(10L, log.segments.activeSegment.baseOffset)
-    val expected = log.segments.values.asScala.toVector
-    val deleted = log.truncateFullyAndStartAt(10L)
+    val expected = new util.ArrayList(log.segments.values)
+    val deleted = new util.ArrayList(log.truncateFullyAndStartAt(10L))
```

Comment:
Why is this needed?

Reply:
They give different types like java `ConcurrentSkipListMap.Values` and `ArrayList`, and the equality check will fail. Wrapped around the same ArrayList instance for their equality checks. 
Followed similar checks in LocalLogTest and UnifiedLogTest.